### ISSUE TITLE
SQLite's LONGVARCHAR datatype is not supported

### ISSUE THREAD

**Issue by ericpgreen2** [duckdblabs/sqlite_scanner#14]
I get `Invalid Error: "LONGVARCHAR"` when selecting from "LONGVARCHAR" columns in an attached SQLite database. To reproduce:

Create a SQLite database:
```bash
echo "CREATE TABLE tbl (long_strings LONGVARCHAR); INSERT INTO tbl VALUES ('abc'), ('def');" | sqlite3 test.db
```

Attach it from DuckDB (`./build/release/duckdb -unsigned`):
```sql
D LOAD 'build/release/extension/sqlite_scanner/sqlite_scanner.duckdb_extension';
D CALL sqlite_attach('test.db');
D SELECT * FROM tbl;
Error: Invalid Error: "LONGVARCHAR"
```

**Comment by ericpgreen2** [comment]
Update: I worked around this by editing the SQLite database's schema to use the `TEXT` type:
```sql
PRAGMA writable_schema = 1;
UPDATE sqlite_master SET sql = replace( sql, 'LONGVARCHAR', 'TEXT' ) WHERE sql LIKE '%LONGVARCHAR%';
PRAGMA writable_schema = 0;
```

**Comment by Alex-Monahan** [comment]
> Update: I worked around this by editing the SQLite database's schema to use the `TEXT` type:
> 
> ```sql
> PRAGMA writable_schema = 1;
> UPDATE sqlite_master SET sql = replace( sql, 'LONGVARCHAR', 'TEXT' ) WHERE sql LIKE '%LONGVARCHAR%';
> PRAGMA writable_schema = 0;
> ```

Neat trick! Thanks for posting the workaround!

**Comment by Mause** [comment]
Had a look into this - turns out SQLite doesn't actually have a `LONGVARCHAR` type, it's just super relaxed on the type names it accepts when a table isn't in "STRICT" mode.

There's more details in the SQLite docs (https://www.sqlite.org/datatype3.html#type_affinity), but the summary is, in non-strict mode, SQLite treats any column with a type name that contains "CHAR"/"CLOB"/"TEXT" as a `TEXT` type (and similar for other types too)

Obviously the error message here needs to be improved, and we possibly need to support this non-strict mode of operation

**Comment by Mytherin** [comment]
Nice find! I think we should duplicate the documented type affinity rules [here](https://github.com/duckdblabs/sqlite_scanner/blob/main/sqlite_scanner.cpp#L109).

**Comment by Mytherin** [comment]
Also considering SQLite allows arbitrary types, we shouldn't copy and paste the text into a SQL query :), e.g .this is valid in SQLite:

```sql
CREATE TABLE sql_injection(i "int); DROP TABLE customers");
```

**Comment by Mause** [comment]
For our future reference, SQLite implements it's affinity logic in the `sqlite3AffinityType` function (through some very odd bit shifting and pointer arithmetic logic)

**Comment by Mytherin** [comment]
Thanks for the report! This should be fixed by #18 now.

**Comment by ericpgreen2** [comment]
Thanks – works great!
